packages/scaffolding-cli/templates/build/azDevops/azure/azure-pipelines-netcore-k8s.yml

```diff
@@ -231,62 +229,61 @@ stages:
                         env_vars: {
-                          namespace: dev-stacks-api,
-                          dns_pointer: api.$(base_domain),
+                          dns_pointer: %REPLACE_ME_FOR_DNS_RECORD%.$(base_domain),
                           tls_domain: $(base_domain),
                           k8s_app_path: /api/menu,
+                          log_level: "Debug",
                           k8s_image: '$(k8s_docker_registry)/$(docker_image_name):$(docker_image_tag)',
-                          version: $(docker_image_tag),
-                          role: $(role),
-                          company: $(company),
-                          project: $(project),
-                          domain: $(domain),
-                          component: api,
-                          app_name: yumido-api,
-                          resource_def_name: xx-replace-me,
-                          environment: dev,
                           aadpodidentitybinding: stacks-webapp-identity,
-                          APPINSIGHTS_INSTRUMENTATIONKEY: $(APPINSIGHTS_INSTRUMENTATIONKEY),
-                          cosmosdb_key: $(COSMOSDB_KEY_DEV)
-                        }
+                          app_insights_key: $(APPINSIGHTS_INSTRUMENTATIONKEY),
+                          cosmosdb_key: $(cosmosdb_primary_master_key),
+                          cosmosdb_endpoint: $(cosmosdb_endpoint),
+                          cosmosdb_name: $(cosmosdb_database_name)
+                        },
+                        args: "-no-empty", # -no-unset is present by default
                       }
                     ]
 
                 - template: azDevOps/azure/templates/v2/steps/deploy-k8s-app-kubectl.yml@templates
                   parameters:
-                    environment: "dev"
                     scripts_dir: $(Agent.BuildDirectory)/s/stacks-pipeline-templates/azDevOps/azure/templates/v2/scripts
-                    download_deploy_artefact: true
-                    deploy_artifact_name: $(build_artifact_deploy_name)
                     test_artefact: "tests"
                     test_baseurl: ""
                     functional_test: false
                     performance_test: false
                     smoke_test: false
                     # Kubernetes Config
-                    kubernetes_clusterrg: $(aks_clusterrg)
-                    kubernetes_clustername: $(aks_clustername)
+                    kubernetes_clusterrg: amido-stacks-nonprod-demo
+                    kubernetes_clustername: amido-stacks-nonprod-demo
                     # Mutation Information
-                    kubectl_filepath: $(Agent.BuildDirectory)/s/$(self_repo)/deploy/k8s/app/api-deploy.yml
-                    kubectl_working_dir: $(Agent.BuildDirectory)/s/$(self_repo)/deploy/k8s/app # unnecessary
-                    appinsights_enabled: false
-                    appinsights_accountName: ""
-                    secureFile: ""
+                    kubectl_filepath:
```

Comment:
@dnitsch just be careful of backwards compatibility for other pipelines using these tasks. Change the param type to a list breaks the other pipelines.

Reply:
so generally the public facing generate files should all have this 
`ref: refs/tags/v1.x.x`  and the breaking change was lumped into a `v.1.3.0` tag so as long as all our current tests are using a version pegged to ~>v1.2 then it shouldn't be a problem